### ISSUE TITLE
Formats

### ISSUE THREAD

**Issue by jonathanrobie** [Copenhagen-Alliance/scripture-burrito#2]
The README.md says a burrito is:
> a single container format to span the entire Scripture life cycle, from translation, through community testing and checking, to publication

But it also describes a burrito thus:
> a single container format to span the entire Scripture life cycle, from translation, through community testing and checking, to publication

Which is it?  If it is a format then it has a physical form that can be read by various platforms.  It might be a zip file or a .jar file or whatever, and which we choose may not greatly matter, but software needs to know what it should be able to accept or produce for exchange.

**Comment by mvahowe** [comment]
@jonathanrobie I'm almost certain that your two quotes are identical.

Regardless, "format" may not be the best word, but DBL has been moving away from delivering "DBL bundles" as a shrink-wrapped zip file, mainly because audio entries can be 15Gb and an entire HD video Bible would be closer to 1Tb. So, personally, I'd prefer to define the recipe for building a burrito and allow people to package it (or not) as they like.

If we're going to pick one optional format, I'd suggest zip, because that's what has been used until now.

Could we change "format" to "structure" or "architecture" or "data model"?

(My precedent for a carefully-defined thingey that doesn't get shipped as such is https://www.w3.org/TR/xpath-datamodel-31/ .)

**Comment by jonathanrobie** [comment]
I don't think this is a data model, it specifies specific types like USFM and USX, even giving version numbers.  The XPath data model represents the parsed form of XML - I doubt we are trying to define something like that for the text files we use.  

I do think we need to specify directory structures or at least a catalog that can map to such a structure.

The multimedia issue is real, and important for interchange.  I suspect we need a story for how we exchange these things, and multimedia generally won't be part of a container for text data and metadata.  But I still suspect we will need to specify real file formats.

**Comment by mvahowe** [comment]
Given that, at this point, DBL treats 5 different types of media in essentially the same way, we need something that works across all those media (plus the media we haven't thought about yet). If we're going for a radical rethink, I think we should consider something like tar, since, for text, compression isn't a big deal (especially when most Internet technologies do gzip on the fly) and, for binary files, compression makes no sense anyway.

**Comment by jonathanrobie** [comment]
I think the Web notion of media types is helpful here - we need a system that handles a known set of media types, is extensible so that other kinds can be added, and foresees the addition of new media types in the future.

tar, zip, etc. are good for directory structures and text files. 

We don't have to say *how* these things are exchanged, just what is exchanged.  Systems may use very different approaches to exchange multimedia, but if we can say what is to be transferred,  I think we have done what this specification needs to do.   A specification for an exchange protocol may also be important, but I think that is outside the scope of this group.

**Comment by mvahowe** [comment]
I think we may be talking past each other regarding "media". In DBL-Metadata-speak, a medium isn't like a mime-type, it's one flavour of entry, containing multiple files of multiple mime-types. So, eg, "audio" isn't an mp3 file, it's a collection of resources, probably in mp3 format, plus metadata, plus maybe other files in other formats. "text" contains many mimetypes including, sometimes, fonts. "print" contains PDFs, but also a JPEG. etc.

**Comment by mvahowe** [comment]
Also, this is a SB issue, not a DBL Metadata 2.3 issue, because DBL isn't going to change how it delivers content for v2.3. (We can still talk about it though...)

**Comment by mvahowe** [comment]
Just to complete my previous example, I would expect Biblica's TTT (InDesign XML) to go in a "print" entry, even though the mimetype and indeed the substantive content is quite different to a rendered page, because they "do" something similar.

**Comment by mvahowe** [comment]
@jonathanrobie This is what the DBL website download page currently looks like. Previously the only option was to download a zip, and going back to that wouldn't be considered a feature by DBL users.

![entry_download](https://user-images.githubusercontent.com/1576594/53097679-53a51a00-3522-11e9-87a1-acedb205bce1.png)
